### PR TITLE
Decomp Interactable "Spring"

### PR DESCRIPTION
**Pull request by JaceCear**

### CODE REVIEW

File: include/sprite.h [hunk 1]
```diff
@@ -19,6 +19,16 @@
 #define SpriteGetScreenPos(spritePos, regionPos)                                        \
     ((spritePos)*TILE_WIDTH + (regionPos)*CAM_REGION_WIDTH)
 
+#define CAM_BOUND_X ((DISPLAY_WIDTH) + (CAM_REGION_WIDTH))
```
Comment (freshollie) [inline include/sprite.h:22]: As discussed, probably move to `game.h` for now and we'll find a proper place later. Same with `SpriteGetScreenPos` and others relating to Entities now thinking about it

File: src/interactable_spring.c [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    = { SE_MUSIC_PLANT_SPRING_1, SE_MUSIC_PLANT_SPRING_2, SE_MUSIC_PLANT_SPRING_3,
+        SE_MUSIC_PLANT_SPRING_4, MUS_DUMMY };
+
+// Finished, but doesn't match.
+static void initSprite_Interactable_Spring(u8 springType, Interactable *inEntity,
+                                           u16 spriteRegionX, u16 spriteRegionY,
+                                           u8 spriteY)
+{
+    s16 springKind = SPRING_KIND_NORMAL;
+    struct Task *t = TaskCreate(&Task_Interactable_Spring, sizeof(Sprite_Spring), 0x2010,
```
Comment (freshollie) [inline src/interactable_spring.c:102]: Btw you don't need to explicitly pass the functions as pointers as this is implicit

File: src/interactable_spring.c [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+    if (sub_800E490(main, entity, spring, &gPlayer) != 0) {
+        gCurTask->main = &sub_800E3D0;
+        main->unk20++;
+
+        if ((LEVEL_TO_ZONE(gCurrentLevel) == ZONE_3 && (spring->unk3D / 2) == 0))
+            main->unk4 = (void *)(OBJ_VRAM0 + 0x2B00);
+    }
+
+#ifndef NON_MATCHING
+    if (IS_OUT_OF_CAM_RANGE(main->unk16, TO_TYPE(s16, main, 12)))
```
Comment (freshollie) [inline src/interactable_spring.c:169]: `unkC` of `UNK_0808B3FC_UNK240` is a pointer to `UNK_0808B3FC_UNK240_UNKC` which is only really used in `sprite_3.c`

You may find the use of `unk1` can be written as `unkC->unk1 >> 8` and then you can convert this to `main->unk0` (With `unk0` being `s16`)

Comment (freshollie) [reply src/interactable_spring.c:169]: Oh wait, this is `unk18` 🤔

File: src/multiplayer_singlepak_results.c [hunk 4]
```diff
@@ -177,7 +177,7 @@ void sub_808207C(void)
                 element->unk18 = 0;
                 element->unk1A = 0x100;
                 element->unk8 = 0;
-#ifndef NONMATCHING
+#ifndef NON_MATCHING
```
Comment (freshollie) [inline src/multiplayer_singlepak_results.c:180]: Nice

File: src/interactable_spring.c (hunk 3 above)
Comment (JaceCear) [reply src/interactable_spring.c:169]: I tried that before it matched, will have a look

File: src/multiplayer_singlepak_results.c (hunk 4 above)
Comment (JaceCear) [reply src/multiplayer_singlepak_results.c:180]: I was just wondering how you wrote that before, so I searched for both and found this was the only one without the underscore!

File: src/interactable_spring.c (hunk 2 above)
Comment (JaceCear) [reply src/interactable_spring.c:102]: It's just the style I got myself accustomed to. But if it's usually not done in your codebase, I'll let it be.

Comment (freshollie) [reply src/interactable_spring.c:102]: Not done yet, I agree it's something which may provide some context, but for now I've not been passing with the `&`

File: src/interactable_spring.c (hunk 3 above)
Comment (freshollie) [reply src/interactable_spring.c]: The code you have should produce the same result :/

Comment (JaceCear) [reply src/interactable_spring.c]: Yeah, it worked after matching! :)

**Review by freshollie — APPROVED** [review]

**Pull request merged**